acceptancetests/utility.py

```diff
@@ -436,11 +440,19 @@ def assert_dict_is_subset(sub_dict, super_dict):
     :raises JujuAssertionError: when sub_dict items are missing.
     :return: True when when sub_dict is a subset of super_dict
     """
-    if not all(item in super_dict.items() for item in sub_dict.items()):
+    if not is_subset(sub_dict, super_dict):
```

Comment:
The original code didn't really work as intended (see checks in assess_storage). This is a recursive variant that does the trick.